implement a bunch of validation keywords

frederikcreemers opened the pull request:
I implemented a bunch of validators from the spec. Mostly object validators, which are hard to define inside tags, and array validators, some of which I might still ad, are still missing.

frederikcreemers committed:
implement a bunch of validation keywords

urakozz committed:
Update vcs.xml

urakozz committed:
Update go-json-schema-generator.iml

urakozz committed:
Update GOPATH__go_json_schema_generator_.xml

urakozz committed:
Update workspace.xml

frederikcreemers commented:
Whoops, that workspace stuff wasn't meant to be in there. Gogland, the IDE I use, adds it, and I usually add it to my .gitignore so it doens't get into git.

urakozz commented:
@bigblind could you please add it to the .gitignore and remove?

frederikcreemers commented:
That's the plan, will do asap.

On Mon, Jul 10, 2017 at 2:40 PM Yury Kozyrev <notifications@github.com>
wrote:

> @bigblind <https://github.com/bigblind> could you please add it to the
> .gitignore and remove?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/urakozz/go-json-schema-generator/pull/1#issuecomment-314093313>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAtA1dG7c4wfTuzQjQJQihuKLoq_ZcLtks5sMhurgaJpZM4ORqxD>
> .
>

frederikcreemers committed:
Ignore IntellIJ Gogland project configuration files.

frederikcreemers committed:
Remove IntellIJ Gogland project files

frederikcreemers committed:
go fmt

yurykozyrev commented on generator.go:
```diff
@@ -139,6 +161,94 @@ func (p *property) readFromStruct(t reflect.Type) {
+
+// Some helper functions for not having to create temp variables all over the place
+func int64ptr(i interface{}) *int64 {
+	j := reflect.ValueOf(i).Convert(reflect.TypeOf(int64(0))).Interface().(int64)
```
Please create common variables  the beginning of the file for reflect types (int64, float64)
```
var rTypeInt64, rTypeFloat64 = reflect.TypeOf(int64(0)), reflect.TypeOf(float64(0))

```
Then in this method please first check if given interface is convertible to the destination you want, like `if value.Type().ConvertibleTo(rTypeInt64) ...`

frederikcreemers replied:
What would you want me to do if the interface is not convertible? Just panic? The generator currently doesn't have a mechanism for returning errors, and in order to add that, I'd have to change the signature of every method so that it can return an error.

urakozz replied:
ignore and return nil.

yurykozyrev commented on generator.go:
```diff
@@ -139,6 +161,94 @@ func (p *property) readFromStruct(t reflect.Type) {
+}
+
+func float64ptr(i interface{}) *float64 {
+	j := reflect.ValueOf(i).Convert(reflect.TypeOf(float64(0))).Interface().(float64)
```
same here

yurykozyrev commented:
@bigblind I added some comments please check. After those fixes let's merge it

frederikcreemers committed:
Improve int64ptr and float64ptr functions

These functions now use global variables to store the reflect.Types they
need, and check whether the values they are given can be converted to the
type in the function name.
